examples/jsm/modifiers/SimplifyModifier.js

```diff
@@ -109,22 +109,63 @@ class SimplifyModifier {
 
 		}
 
+		// sort & reindex now.
+		// maintaining order by edge collapse cost will be cheaper
+		// than searching for lowest every time.
+		// Lowest cost at tail will minimize reindexing on removal.
+		vertices.sort( ( a, b )=>{
+
+			return b.collapseCost - a.collapseCost;
+
+		} );
+		for ( let i = 0, il = vertices.length; i < il; i ++ ) {
+
+			vertices[ i ].id = i;
+
+		}
+
 		let nextVertex;
 
 		let z = count;
 
-		while ( z -- ) {
+		// likely only a minor optimization,
+		// but why check for threshold every iteration
+		// in cases when we aren't even using it?
+		if ( threshold ) {
```

Comment:
Is `threshold` a boolean?

Reply:
Good question- no it is not.  If provided, it is a numeric value in whatever units collapseCost is in, but it is intended to be optional so as not to break existing usage.  A value of 0 would, moreover, result in no actual vertex removal (the first vertex found would already exceed the threshold), in which case the same behavior could be achieved by setting count to 0, and in my initial conception, this was consequently a safe case to ignore.   If `typeof( threshold ) == 'number' && threshold >= 0` is a more consistent, efficient, or readable-per-unit-efficiency check than relying on js truthy resolution of numeric values, that's a quick change.

Reply:
Ah, I think `if ( threshold >= 0 )` should do the trick then.